specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json

```diff
@@ -1182,6 +1182,36 @@
         }
       }
     },
+    "TeradataTableDatasetTypeProperties": {
+      "description": "Teradata dataset properties.",
+      "properties": {
+        "database": {
+          "type": "object",
+          "description": "The database name of Teradata. Type: string (or Expression with resultType string)."
+        },
+        "table": {
+          "type": "object",
+          "description": "The table name of Teradata. Type: string (or Expression with resultType string)."
+        }
+      }
+    },
+    "TeradataTableDataset": {
+      "x-ms-discriminator-value": "TeradataTable",
+      "description": "The Teradata database dataset.",
+      "type": "object",
```

Comment:
>object [](start = 15, length = 6)

I see that you're modeling a lot of properties as object without referencing a definition. In the c# generated SDK, the property type will be `object`. Is that intended? How do you pass an expression via the C# SDK? What kind of expression are these also?

Reply:
Discussed on a call. C# SDK has a class that users can new up and pass in for the property value. Also backend supports multiple types.

---
In reply to: [295494222](https://github.com/Azure/azure-rest-api-specs/pull/6346#discussion_r295494222) [](ancestors = 295494222)